packages/create-cloudflare/src/helpers/command.ts

```diff
@@ -16,55 +16,90 @@ type RunOptions = {
 	cwd?: string;
 };
 
+type PrintOptions<T> = {
+	promise: Promise<T> | (() => Promise<T>);
+	useSpinner?: boolean;
+	startText: string;
+	doneText?: string;
+};
+
 export const runCommand = async (
 	command: string,
 	opts?: RunOptions
 ): Promise<string> => {
-	const s = spinner();
+	return printAsyncStatus({
+		useSpinner: opts?.silent, // only use the spinner if we're not spawning with stdio: 'inherit'
+		startText: opts?.startText || command,
+		doneText: opts?.doneText,
+		promise() {
+			const [executable, ...args] = command.split(" ");
+
+			const squelch = opts?.silent || process.env.VITEST;
+
+			const cmd = spawn(executable, [...args], {
+				// TODO: ideally inherit stderr, but npm install uses this for warnings
+				// stdio: [ioMode, ioMode, "inherit"],
+				stdio: squelch ? "pipe" : "inherit",
+				env: {
+					...process.env,
+					...opts?.env,
+				},
+				cwd: opts?.cwd,
+			});
+
+			let output = ``;
+
+			if (opts?.silent) {
```

Comment:
Should this be?

```suggestion
			if (squelch) {
```

Reply:
I did change the to squelch but then changed it back because it looked intentional. squelch was used to suppress stdin/stdout during tests, but collecting output is only needed if the implementation code set silent – that would be changing the return value in test environments.

Now thinking about it again, stderr would be useful in test output.

cc @jculvey any reason not to change this to `squelch`?